### PR TITLE
Buildbot POJO

### LINKED ISSUE #12: Create Interface between Parser and Storage Controller
Create endpoints for Parsers to POST information to datastore via Storage Controller.

Requirement Reference: 4

### PR DESCRIPTION
**Pull request by MagoDelBlocco**
Class containing the information coming from the Buildbot Fetchers.
It is supposed to contain the compilation info for an individual revision, from a build bot. The data included should be the builder's name, status, logs, and the commit hash of the revision which has been compiled.
The instances should be immutable, with an external (not storage) component being able to create them, and only the storage should be able to consume them.

Partially closes #12 along with #44

### CODE REVIEW

File: src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    this.builderName = builderName;
+    this.logs = new LinkedList<>(logs);
+    this.status = status;
+  }
+
+  /**
+   * Checks for the existance of the fields necessary for updating a database entry.
+   * Does not check the validity of each field.
+   *
+   * @return true if all required fields are available, false if not.
```
Comment (samuelsogbesan) [inline src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java:55]: nit
```suggestion
   * @return True if all required fields are available, False if not.
```

**Review by samuelsogbesan — APPROVED** [review]
LGTM, one suggested nit.

File: src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java (hunk 1 above)
Comment (MForster) [reply src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java:55]: lower case is ideomatic

File: src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.graphgeckos.dashboard.storage;
+
+/**
+ * An immutable container used for providing stripped-down build bot information
+ * to the StorageController. Provides functionality for checking whether the information
```
Comment (Somae) [inline src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java:19]: Please correct "StorageController".

File: src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+ * is sufficient for an update of an entry. Does not modify or validity check the individual
+ * contents of each field.
+ *
+ * Fields included: 
+ * -commitHash: the commit hash of the revision built by a certain build bot.
+ * -builderName: the name of the build bot which performed the compilation.
+ * -logs: name of the stage and links to the logs for each stage of the compilation.
+ * -status: the final result of the compilation (passed or failed).
```
Comment (Somae) [inline src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java:27]: Aren't there also build that never finish?

File: src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+ */
+public class ParsedBuildbotData {
+  private final String commitHash;
+  private final String builderName;
+  private final List<Log> logs;
+  private final boolean status;
+
+  /**
+   * Constructs an immutable instance of a ParsedBuildbotData.
```
Comment (Somae) [inline src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java:36]:
```suggestion
   * Constructs an immutable instance of ParsedBuildbotData.
```

File: src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+   *
+   * @param commitHash the commit hash of the revision this data refers to
+   * @param builderName the name of the builder which performed the compilation
+   * @param logs logs for each individual stage of compilation. See {@link
+   *      #com.google.graphgeckos.dashboard.storage.Log Log}
+   * @param status the results of the compilation (True for passed, False for failed)
```
Comment (Somae) [inline src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java:42]: Please change to lowercase (reference Mike's review).
```suggestion
   * @param status the results of the compilation (true for passed, false for failed)
```

File: src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    this.builderName = builderName;
+    this.logs = new LinkedList<>(logs);
+    this.status = status;
+  }
+
+  /**
+   * Checks for the existance of the fields necessary for updating a database entry.
+   * Does not check the validity of each field.
+   *
+   * @return True if all required fields are available, False if not.
```
Comment (Somae) [inline src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java:56]:
```suggestion
   * @return true if all required fields are available, false if not.
```

**Review by Somae — APPROVED** [review]
LGTM with a few nits.

File: src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java (hunk 3 above)
Comment (MagoDelBlocco) [reply src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java:27]: Is it possible for a worker to start compiling a revision and then abandon it?

Comment (Somae) [reply src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java:27]: I am not entirely sure if all buildbots can, but I have seen some where this is the case. Some build bots abandon old revisions in favor of a long queue of newer revisions for example. I only want you to be cautious about assuming that the status is always binary, unless you are discarding everything else, though that might make combining data from different bots a challenge.

Comment (MagoDelBlocco) [reply src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java:27]: Changed the field to "BuildStatus" type, an enum which can be easily expanded, from PR #36

**Pull request merged**